contracts/LightClientBridge.sol

```diff
@@ -213,14 +220,14 @@ contract LightClientBridge {
             // @note Type conversion from bytes32 -> uint8, by way of bytes1 (to work around limitations)
             onChainRandNums[i] = uint8(bytes1(randomSeedBlockHash));
         }
+        // TODO this might lead to duplicate entries?
```

Comment:
The random number generator above could yield duplicate positions. As mentioned elsewhere, it's also quite inefficient not working with a bitfield here. 

I'd actually propose to just hash some data into a word, and then add/remove bits until the right number of bits are set.